### DISCUSSION TITLE
is my repository still pretty much private even if made public so i can talk to people about it?

### DISCUSSION THREAD

**Discussion by Larwin64** [community/community#135455] [Repositories] [answered]
### Select Topic Area

Question

### Body

im learning git hub by trial and error seem my repository is still hidden even though it public.,

**Comment by TheGiraffe3** [comment]
Just because it's a public repository doesn't always mean people will come look at it.
I can see your repository (assuming there is only one), and I'm sure other people can as well.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Larwin64 !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Repositories category. I've gone ahead and moved it for you. Good luck!

**Reply by Larwin64** [reply]
thank you

**Reply by LiteBrite82** [reply, marked as answer]
Hi @Larwin64

Welcome to GitHub!!!! 

To answer your question

> is my repository still pretty much private even if made public

It depends on what you mean by private. 

Short answer: 
anyone can see and make copies of your "public" repos; no one can change the code in your repo without you giving them access - making them a "collaborator".

Long answer:
non collaborators can create issues in your repository and "fork" your repository (make their own copy that is linked to your repo).

Folks can use "forks" and "pull requests" to make suggested changes to your repo that you would have to approve if you wanted to integrate the suggested change into your code base.  

There are a number of settings you can change to limit other user's ability to interact with your repositories.

**Reply by LiteBrite82** [reply]
wether public or private, never put sensitive information in a repo - Personal information, API Keys, passwords etc. 
`.gitignore` files are crucial to a repositories security. It is a simple file you keep in the root of your repo that lists specific files that you do not want uploaded to the cloud. So even if you commit and push using `git add all`; any file listed in `.gitignore` will not be added to the cloud. [Gitignore documentation](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files)

**Comment by LiteBrite82** [comment]
Are there any other aspects of GitHub you have questions about?

**Reply by Larwin64** [reply]
guess im understanding the repository is a seperate channels than the more public channels i dont share personal info that much online wich is why i dont go on facebook as much as i used to.

**Reply by Larwin64** [reply]
so i have give them permission to download or change anything with pull requests in mind?

**Reply by Larwin64** [reply]
can people see code or my text files?

**Reply by Larwin64** [reply]
or make suggestions?

**Reply by Larwin64** [reply]
not sure i need share files that much as learning to be a better programmer i have found certain things recently i have to do in my code or the final product wont go as attended like with mathematics you need to decide if you want both positive and negative values in the final product cause im finding this may cause unwanted behavior in my projects.

**Reply by Larwin64** [reply]
im not even pre alpha far as project i started goes as im barely into it im learning how programming works.

**Reply by Larwin64** [reply]
im finding backing up my project is good idea why i chose github i never heard word 'repository' before github.

**Reply by Larwin64** [reply]
is there a way get advice for free on learning to code better?

**Reply by Larwin64** [reply]
you have to build the sailboat before you sail with it and its more than 2 planks nailed together. :)

**Reply by Larwin64** [reply]
i made public cause it said i had to to chat about my project.

**Reply by Larwin64** [reply]
can i still access the comunity if i go private again?

**Reply by Larwin64** [reply]
im a student for life

**Reply by LiteBrite82** [reply]
You can still access everything if you go private. if you want to have someone collaborate on a project with you, you would have to give them access to even see the repository.

**Comment by LiteBrite82** [comment]
what programming languages are you using / learning / interested in?
